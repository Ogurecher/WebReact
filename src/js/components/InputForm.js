import React, { Component } from "react";

export default class InputForm extends Component {
    constructor(props) {
        super(props);
        this.state = {value: ''};

        this.handleChange = this.handleChange.bind(this);
        this.handleSubmit = this.handleSubmit.bind(this);
    }

    handleChange(event) {
        this.setState({value: event.target.value});
    }

    handleSubmit(event) {
        //TODO add to favorites
        alert(this.state.value);
        event.preventDefault();
    }

    render() {
        return (
            <form onSubmit={this.handleSubmit}>
                <input type='text' value={this.state.value} onChange={this.handleChange} />
                <input type='submit' value='Add to Favorites' />
            </form>
        );
    }
}