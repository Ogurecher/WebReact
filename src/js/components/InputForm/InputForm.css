.favourites-form-wrapper {
    margin: 1%;
    display: flex;
    flex-direction: row;
    align-items: baseline;
}

.favourites-text {
    font-size: 200%;
    margin-left: 1%;
    margin-right: 1%;
}