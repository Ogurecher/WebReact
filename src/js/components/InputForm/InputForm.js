import React, { Component } from 'react';
import { addToFavourites } from '../../actions/index';
import { connect } from 'react-redux';
import './InputForm.css';

export class InputForm extends Component {
    constructor(props) {
        super(props);
        this.state = {value: ''};

        this.handleChange = this.handleChange.bind(this);
        this.handleSubmit = this.handleSubmit.bind(this);
    }

    handleChange(event) {
        this.setState({value: event.target.value});
    }

    handleSubmit(event) {
        event.preventDefault();
        
        this.props.addToFavourites(this.state.value);
        this.setState({value: ''});
    }

    render() {
        return (
            <form onSubmit={this.handleSubmit} className='favourites-form-wrapper'>
                <p className='favourites-text'>Favourites</p>
                <input type='text' value={this.state.value} onChange={this.handleChange} />
                <input type='submit' value='+' />
            </form>
        );
    }
}

function mapStateToProps(state) {
    return {};
}

export default connect(
    mapStateToProps,
    { addToFavourites }
  )(InputForm);