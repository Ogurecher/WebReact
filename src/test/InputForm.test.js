import React from 'react';
import renderer from 'react-test-renderer';
import configureMockStore from 'redux-mock-store';
import { Provider } from 'react-redux';

import InputForm from '../js/components/InputForm/InputForm';

const store = configureMockStore();

describe('InputForm', function() {
    test('snapshot renders', function() {
        const storeMock = store();

        const component = renderer.create(<Provider store={storeMock} ><InputForm /></Provider>);
        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });
});