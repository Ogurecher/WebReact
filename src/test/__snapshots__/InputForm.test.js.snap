// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputForm snapshot renders 1`] = `
<form
  className="favourites-form-wrapper"
  onSubmit={[Function]}
>
  <p
    className="favourites-text"
  >
    Favourites
  </p>
  <input
    onChange={[Function]}
    type="text"
    value=""
  />
  <input
    type="submit"
    value="+"
  />
</form>
`;
